plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftUnloadSplitSource.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.redshift;
+
+import com.amazon.redshift.util.RedshiftException;
+import com.google.common.collect.ImmutableList;
+import io.airlift.log.Logger;
+import io.trino.plugin.jdbc.JdbcSplit;
+import io.trino.spi.connector.ConnectorSplit;
+import io.trino.spi.connector.ConnectorSplitSource;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class RedshiftUnloadSplitSource
+        implements ConnectorSplitSource
+{
+    private static final Logger log = Logger.get(RedshiftUnloadSplitSource.class);
+    public static final String RESULT_FILES_QUERY = "SELECT rtrim(path) FROM stl_unload_log WHERE query=pg_last_query_id() ORDER BY path";
```

Comment:
Cluster could have multiple workloads right ? How can we ensure the `pg_last_query_id` is the correct one to fetch the location ?

Reply:
[The docs](https://docs.aws.amazon.com/redshift/latest/dg/PG_LAST_QUERY_ID.html) say (emphasis mine):
> Returns the query ID of the most recently completed query **in the current session**.

We do use separate connections for every Trino query, so I think it should be fine.